src/Native/Unix/CMakeLists.txt

```diff
@@ -247,6 +240,14 @@ endfunction()
 include(configure.cmake)
 
 add_subdirectory(System.IO.Compression.Native)
+
+add_compile_options(-Weverything)
+add_compile_options(-Wno-c++98-compat-pedantic)
+add_compile_options(-Werror)
```

Comment:
We should keep warnings as errors by default for everything so that they do not creep up. It is annoying to look at the warnings during builds.